tokio-threadpool/src/lib.rs

```diff
@@ -3,6 +3,107 @@
 #![doc(html_root_url = "https://docs.rs/tokio-threadpool/0.1.4")]
 #![deny(warnings, missing_docs, missing_debug_implementations)]
 
+// The Tokio thread pool is a thread pool designed to scheduled futures in Tokio
+// based applications. The thread pool structure manages two sets of threads:
+//
+// * Worker threads.
+// * Backup threads.
+//
+// Worker threads are primary threads and are used to schedule futures. These
+// threads operate using a work-stealing strategy. Backup threads are to support
+// the `blocking` API. Threads will transition between the two sets.
+//
+// The advantage of the work-stealing strategy is minimal cross thread
+// coordination. The thread pool attempts to make as much progress as possible
+// without communicating across threads.
+//
+// # Crate layout
+//
+// The primary type is `Pool`. This struct contains the majority of the pool
+// state. Because worker threads will shutdown and be reestarted, the pool also
+// contains the state for each worker. The state dedicated to a worker is
+// represented by `worker::Entry`.
+//
+// `Worker` contains the logic that runs on each worker thread. It holds an
+// `Arc` to `Pool` and is able to access its state from `Pool`.
+//
+// `Task` is a harness around an individual future. It manages polling and
+// scheduling that future.
+//
+// # Worker overview
+//
+// Each worker has two queues: a deque and a mpsc channel. The deque is the
+// primary queue for tasks that are scheduled to run on the worker thread. Tasks
+// can only be pushed onto the deque by the worker, but other workers may
+// "steal" from that deque. The mpsc channel is used to submit futures while
+// external to the pool.
+//
+// As long as the thread pool has not been shutdown, a worker will run in a
+// loop. Each loop, it consumes all tasks on its mpsc channel and pushes it onto
+// the deque. It then pops tasks off of the deque and executes them.
+//
+// If a worker has no work, i.e., both queues are empty. It attempts to steal.
+// To do this, it randomly scans other workers' deques and tries to pop a task.
+// If it finds no work to steal, the thread goes to sleep.
+//
+// # Thread pool initialization
+//
+// By default, no threads are spawned on creation. Instead, when new futures are
+// spawned, the pool first checks if there are enough active workeer threads. If
+// not, a new worker thread is spawned.
+//
+// # Spawning futures
+//
+// The process for spawning a future depends on whether the action is taken from
+// a workere thread or external to the thread pool.
```

Comment:
"a worker thread"

better: The spawning behavior depends on whether a future was spawned from within a worker or thread or if it was spawned from an external handle.